### PR TITLE
Changes for PR-1 of BlameModeForDump

### PR DESCRIPTION
**Pull request by karanjitsingh**
Goal of BlameModeForDump branch is to add dump facility into the blame data collector.

This is the first PR out of a total of 3 expected PRs
    * We add TestHostInitialized event to the data collection events contract
    * TestHostInitialized will be triggered right after spawning test host process 
    * For a data collector, TestHostInitialized will be fired after SessionStart and before TestCaseStart
    * ProxyExecutionManagerWithDataCollection will subscribe to TestHostManager.HostLaunched for test host process id

Subsequent PRs
    * PR2: Dump utility implementation + tests
    * PR3: Integration + tests

### CODE REVIEW

File: src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs [hunk 1]
```diff
@@ -282,6 +282,19 @@ public void ProcessRequests()
                         }
 
                         break;
+
+                    case MessageType.TestHostInitialized:
+                        if (EqtTrace.IsInfoEnabled)
```
Comment (smadala) [inline src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs]: Add test for this.

File: src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestSender.cs [hunk 2]
```diff
@@ -101,6 +101,12 @@ public void Close()
             this.communicationManager?.StopServer();
         }
 
+        /// <inheritdoc/>
+        public void SendAfterTestHostInitialized(int processId)
+        {
+            this.communicationManager.SendMessage(MessageType.TestHostInitialized, JsonDataSerializer.Instance.Serialize(processId));
```
Comment (smadala) [inline src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestSender.cs]: Use `this.dataSerializer`. Add test for this.

File: src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestSender.cs [hunk 3]
```diff
@@ -101,6 +101,12 @@ public void Close()
             this.communicationManager?.StopServer();
         }
 
+        /// <inheritdoc/>
+        public void SendAfterTestHostInitialized(int processId)
```
Comment (smadala) [inline src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestSender.cs]: Can we rename this to `SendTestHostInitialized`?

File: src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs (hunk 1 above)
Comment (karanjitsingh) [reply src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs]: updated DataCollectionRequestHandlerTests.cs

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManagerWithDataCollection.cs [hunk 4]
```diff
@@ -45,6 +45,13 @@ public ProxyExecutionManagerWithDataCollection(IRequestData requestData, ITestRe
             this.DataCollectionRunEventsHandler = new DataCollectionRunEventsHandler();
             this.requestData = requestData;
             this.dataCollectionEnvironmentVariables = new Dictionary<string, string>();
+
+            testHostManager.HostLaunched += this.TestHostLaunchedHandler;
```
Comment (smadala) [inline src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManagerWithDataCollection.cs]: Add acceptance test for this.

**Comment by singhsarab** [comment]
@dotnet-bot test Windows_NT / Release Build

File: src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs (hunk 1 above)
Comment (karanjitsingh) [reply src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs]: Done

File: src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManagerWithDataCollection.cs (hunk 4 above)
Comment (karanjitsingh) [reply src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManagerWithDataCollection.cs]: Done

**Review by smadala — APPROVED** [review]

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs [hunk 5]
```diff
@@ -251,6 +251,18 @@ public Collection<AttachmentSet> SessionEnded(bool isCancelled = false)
             return new Collection<AttachmentSet>(result);
         }
 
+        /// <inheritdoc/>
+        public void TestHostInitialized(int processId)
+        {
+            // If datacollectors are not configured or datacollection is not enabled, return false.
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs]: >// If datacollectors are not configured or datacollection is not enabled, return false. [](start = 12, length = 87)

Nit:This comment does not make sense

Comment (karanjitsingh) [reply src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs]: removed

**Review by singhsarab — APPROVED** [review]
:shipit:

**Comment by karanjitsingh** [comment]
@dotnet-bot test Windows_NT / Release Build

**Comment by smadala** [comment]
Add documentation https://github.com/Microsoft/vstest-docs/blob/master/docs/extensions/datacollector.md#datacollectionevents for `TestHostInitialized` event.

**Pull request merged**